ctapipe/reco/shower_processor.py

```diff
@@ -0,0 +1,194 @@
+This processor will be able to process a shower/event in 3 steps:
+- shower geometry
+- estimation of energy (optional, currently unavailable)
+- estimation of classification (optional, currently unavailable)
+
+"""
+from traitlets import Bool
+
+from astropy.coordinates import SkyCoord, AltAz
+
+from ctapipe.core import Component, QualityQuery
+from ctapipe.core.traits import List
+from ctapipe.containers import ArrayEventContainer, ReconstructedShowerContainer
+from ctapipe.instrument import SubarrayDescription
+from ctapipe.reco import HillasReconstructor
+
+
+DEFAULT_SHOWER_PARAMETERS = ReconstructedShowerContainer()
+
+
+class ShowerQualityQuery(QualityQuery):
+    """Configuring shower-wise data checks."""
+
+    quality_criteria = List(
+        default_value=[
+            ("> 50 phe", "lambda p: p.hillas.intensity > 50"),
+            ("Positive width", "lambda p: p.hillas.width.value > 0"),
+            ("> 3 pixels", "lambda p: p.morphology.num_pixels > 3"),
+        ],
+        help=QualityQuery.quality_criteria.help,
+    ).tag(config=True)
+
+
+class ShowerProcessor(Component):
+    """
+    Needs DL1_PARAMETERS as input.
+    Should be run after ImageProcessor to produce all DL1b information.
+
+    For the moment it only supports the reconstruction of the shower geometry
+    using ctapipe.reco.HillasReconstructor.
+
+    It is planned to support also energy reconstruction and particle type
+    classification which are now disabled by default.
+    """
+
+    reconstruct_energy = Bool(
+        default_value=False, help="Reconstruct the energy of the event."
+    ).tag(config=True)
+
+    classify = Bool(
+        default_value=False, help="Classify the particle type associated with the event."
+    ).tag(config=True)
+
+    def __init__(
+        self,
+        subarray: SubarrayDescription,
+        config=None,
+        parent=None,
+        **kwargs
+    ):
+        """
+        Parameters
+        ----------
+        subarray: SubarrayDescription
+            Description of the subarray. Provides information about the
+            camera which are useful in calibration. Also required for
+            configuring the TelescopeParameter traitlets.
+        config: traitlets.loader.Config
+            Configuration specified by config file or cmdline arguments.
+            Used to set traitlet values.
+            This is mutually exclusive with passing a ``parent``.
+        parent: ctapipe.core.Component or ctapipe.core.Tool
+            Parent of this component in the configuration hierarchy,
+            this is mutually exclusive with passing ``config``
+        """
+
+        super().__init__(config=config, parent=parent, **kwargs)
+        self.subarray = subarray
+        self.check_shower = ShowerQualityQuery(parent=self)
+        self.reconstructor = HillasReconstructor()
+
+    def _reconstruct_shower(
+        self,
+        event,
+        default=DEFAULT_SHOWER_PARAMETERS,
+    ) -> ReconstructedShowerContainer:
```

Comment:
I think you need to pull in the latest changes from master, since the DL2 PR defined all the classes needed for DL2 storage (and they are no longer called the same things)

Reply:
This is `ReconstructedGeometryContainer` now

Reply:
yes working on it